### PR TITLE
fix: re-process non-image figures through parseFunc_RTE (#698)

### LINKED ISSUE #698: Image in table (v13.6.1)
Thanks for the quick-fix, but it doesen'f fully work as before for img in tables (see #692).

**Figcaption:** doesn't use style="max-width: {image.width}" for the image figure-tag if in table

BE:
```
<figure class="table" style="float:left;width:500px;">
    <table>
        <thead>
            <tr>
                <th>
                    table with img
                </th>
            </tr>
        </thead>
        <tbody>
            <tr>
                <td>
                    <figure class="image image-block">
                        <img src="/fileadmin/user_upload/moto_dog_1.jpg" data-htmlarea-file-uid="2" data-htmlarea-file-table="sys_file" width="112" height="150" title="" alt="" data-title-override="true" data-alt-override="true" data-quality="retina" data-caption="img figcaption for image in table">
                        <figcaption>
                            img figcaption for image in table
                        </figcaption>
                    </figure>
                </td>
            </tr>
        </tbody>
    </table>
</figure>
```
_Annotation: outer figure/table-figure -> figure class="table" style="float:left;width:500px;" style is set for the table, it's not from rte_ckeditor_image_

FE:
`<figure class="table" style="float:left;width:500px;"><table><thead><tr><th>table with img</th></tr></thead><tbody><tr><td><figure class="image image-block"><figcaption>img figcaption for image in table</figcaption></figure></td></tr></tbody></table></figure>`

Image

**No zoom/popup:** No a-tag in FE if enlarge is applied in BE 

Image

BE:
```
<figure class="table" style="float:left;width:500px;">
    <table>
        <thead>
            <tr>
                <th>
                    table with img
                </th>
            </tr>
        </thead>
        <tbody>
            <tr>
                <td>
                    <figure class="image image-left">
                        <img src="/fileadmin/user_upload/moto_dog_1.jpg" data-htmlarea-file-uid="2" data-htmlarea-file-table="sys_file" width="112" height="150" title="" alt="" data-title-override="true" data-alt-override="true" data-htmlarea-zoom="true" data-quality="retina" data-caption="img figcaption for image in table">
                        <figcaption>
                            img figcaption for image in table
                        </figcaption>
                    </figure>
                </td>
            </tr>
        </tbody>
    </table>
</figure>
```

FE:
`<figure class="image image-left"><figcaption>img figcaption for image in table</figcaption></figure>`

**Internal Links:** are not resolved in FE (a href="t3://page?uid=1#1")

`<figure class="table" style="float:left;width:500px;"><table><thead><tr><th>table with img</th></tr></thead><tbody><tr><td><figure class="image image-left"><a href="t3://page?uid=1#1></a><figcaption>img figcaption for image in table</figcaption></figure></td></tr></tbody></table></figure>`

### PR DESCRIPTION
**Pull request by CybotTM**
## Summary

Fixes #698 — Images inside CKEditor 5 table cells weren't rendered properly in the frontend.

**Root cause:** PR #693 (fix for #692) correctly updated `hasFigureWrapper()` XPath to reject table figures, but `renderFigure()` then returned the HTML unchanged, leaving nested `<figure class="image">` inside table cells unprocessed.

**Fix:** When `renderFigure()` receives a non-image figure (table, media, etc.), extract the inner HTML (stripping the outer `<figure>` tags to prevent infinite recursion), re-process it through `parseFunc_RTE`, then re-wrap with the original outer figure tags. This ensures nested image figures, standalone images, and `t3://` links all get properly handled.

**Three symptoms fixed:**
- `max-width` style now applied to inner `<figure>` elements
- Zoom/popup links now wrap images with `data-htmlarea-zoom`
- `t3://` links now resolve to actual URLs

## Changes

- **`ImageRenderingAdapter::renderFigure()`** — Re-process inner content of non-image figures through `parseFunc_RTE` instead of returning unchanged; explicitly set request on cObj before `parseFunc()` call
- **Unit tests** — 3 new tests: inner content reprocessing, attribute preservation, edge case (no cObj)
- **E2E tests** — 6 new frontend rendering tests for table images (zoom, plain, linked, caption, max-width)
- **Demo content** — New CE with table images for DDEV v13/v14 installs
- **Test data** — New CE in `runTests.sh` for E2E test fixtures

_Note: Functional tests for the `parseFunc_RTE` re-processing path were intentionally omitted — they require full TYPO3 frontend bootstrap (TypoScript, TSFE, applicationType) which the functional test framework doesn't provide. Coverage is ensured by unit tests (mock-verify parseFunc args) and E2E tests (actual frontend rendering)._

## Test plan

- [x] Unit tests pass (`composer ci:test:php:unit` — 743 tests, 1744 assertions)
- [x] PHPStan passes (`composer ci:test:php:phpstan` — 0 errors)
- [ ] E2E tests pass on v13 and v14
- [ ] Visual check of DDEV demo page with table images

### CODE REVIEW

File: Tests/Functional/Controller/RteMixedContentRenderingTest.php [hunk 1]
```diff
@@ -1729,4 +1729,127 @@ public function inlineImageInHeadingRendered(): void
...
+        // Zoom image should be wrapped in a link (popup template)
+        self::assertStringContainsString('<a ', $result, 'Zoomable image should have link wrapper');
+        self::assertStringContainsString('Zoomable in table', $result, 'Caption should be preserved');
+    }
+
+    /**
+     * Test: Table figure without nested images returns unchanged.
+     *
+     * A table figure that contains only text cells should pass through
+     * without modification (no image processing needed).
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Tests/Functional/Controller/RteMixedContentRenderingTest.php]: Good catch — updated the docblock to say "preserves table structure" and "keep its table content intact, without any additional image-related processing" which accurately reflects the test behavior.
